docs/user_guide/building_blocks/selectors.md

```diff
@@ -0,0 +1,648 @@
+
+    # use increment stats selector to update dictionary to be saved in json format
+    events = self[increment_stats](events, results, stats, **kwargs)
+
+    return events, results
+```
+
+Notes:
+- If you want to build this exposed {py:class}`~columnflow.selection.Selector` along with the inner
+{py:class}`~columnflow.selection.Selector`s in a new file, you will still need to put the name of
+the new file along with its path in the ```law.cfg``` file under the ```selection_modules```
+argument for law to be able to find the file.
+
+- If you want to use some fields, like the ```Jet``` field, as a Lorentz vector to apply operations
+on, you might use the {py:func}`~columnflow.production.util.attach_coffea_behavior` function. This
+function can be applied on the ```events``` array using
+```python
+events = self[attach_coffea_behavior](events, **kwargs)
+```
+If the name of the field does not correspond to a standard field name, e.g. "BtaggedJets", which
+should provide the same behaviour as a normal jet, the behaviour can still be set, using
+```python
+collections = {x: {"type_name": "Jet"} for x in ["BtaggedJets"]}
+events = self[attach_coffea_behavior](events, collections=collections, **kwargs)
+```
+
+- The actual creation of the weights to be applied in the histogramms after the selection should be
+done in the {py:class}`~columnflow.tasks.production.ProduceColumns` task, using the stats object
+created in this task if needed.
+
+
+
+## Running the SelectEvents task
+
+The {py:class}`~columnflow.tasks.selection.SelectEvents` task runs a specific selection script and
+saves the created masks for event and objects selections in a parquet file, as well as the
+statistics of the selection in a json file.
+
+While it is possible to see all the arguments and their explanation for this task using
+```law run SelectEvents --help```, the only argument created specifically for this task is the
+```--selector``` argument, through which the exposed {py:class}`~columnflow.selection.Selector` to
+be used can be chosen.
+
+The masks created by this task are then used by the
+{py:class}`~columnflow.tasks.reduction.ReduceEvents` task to reduce the number of
+events (see the ```steps``` argument for the {py:class}`~columnflow.selection.SelectionResult`) and
+create/update new columns/fields with only the selected objects (see the ```objects``` argument for
+the {py:class}`~columnflow.selection.SelectionResult`). The saved statistics are used e.g. for the
+weights needed for plotting.
+
+It should not be forgotten that any column created in this task should be included in the
+```keep_columns``` argument of the config file, as only columns explicitely required to be kept in
+this dictionary will be loaded and then saved in the reduced parquet file.
+
+An example of how to run this task for an analysis with several datasets and configs is given below:
+
+```shell
+law run SelectEvents --version name_of_your_version \
```

Comment:
ok changed every time it is called using law run, for now not changed in the text as I fear it would confuse people looking at the API docu more than anything else. I can still change it if you dislike it. On the other hand, it might be a good idea to mention in the law section and/or at the start of the task subsection in the structure section that the cf. prefix needs to be used to run the tasks. Why do you think?

Reply:
You're probabily right since we are linking the actual class name, which does not include the `cf.`prefix. And as you mentioned, shortly mentioning this prefix somewhere would be nice.

Reply:
done, see next commits